### PR TITLE
Updates nightwatch to work with latest haikro

### PR DESCRIPTION
**Pull request by AlbertoElias**

### CODE REVIEW

File: lib/tasks/test.js [hunk 1]
```diff
@@ -146,14 +147,14 @@ module.exports._runBrowserTest = function(config) {
 			configContent = JSON.parse(configContent);
 			configContent.test_settings.default.launch_url = config.testUrl;
 
-			fs.writeFile(nightwatchConfigPath, JSON.stringify(configContent), function(error) {
+			fs.writeFile(nightwatchConfigPath.replace('json', 'tmp.json'), JSON.stringify(configContent), 'utf8', function(error) {
```
Comment (samgiles) [inline lib/tasks/test.js]: Not sure why this change is necessary, but could it be one once on line 138?

File: lib/tasks/test.js [hunk 2]
```diff
@@ -181,7 +184,7 @@ module.exports.browserTest = function(gulp, config) {
 	return Promise.all([
 		process.env.HEROKU_AUTH_TOKEN ? Promise.resolve(process.env.HEROKU_AUTH_TOKEN) : commandLine.run('heroku', ['auth:token']),
 		commandLine.run('git', ['rev-parse HEAD']),
```
Comment (samgiles) [inline lib/tasks/test.js]: Not part of the PR so don't worry too much, but it might be better to be consistent here. Instead of space separating arguments, having them as separate entries into the array.

File: lib/tasks/test.js (hunk 1 above)
Comment (AlbertoElias) [inline lib/tasks/test.js]: What change exactly? I don't want to modify the original file, so I write to a temporary one

File: lib/tasks/test.js (hunk 2 above)
Comment (AlbertoElias) [inline lib/tasks/test.js]: I'll change it here though

File: lib/tasks/test.js (hunk 1 above)
Comment (samgiles) [inline lib/tasks/test.js]: Makes sense, I just mean de-duplicate the logic on line 150 and 157 so you only do this in one place.

**Comment by AlbertoElias** [comment]
@samgiles I've applied the changes

File: lib/tasks/test.js [hunk 3]
```diff
@@ -146,14 +147,16 @@ module.exports._runBrowserTest = function(config) {
 			configContent = JSON.parse(configContent);
 			configContent.test_settings.default.launch_url = config.testUrl;
 
-			fs.writeFile(nightwatchConfigPath, JSON.stringify(configContent), function(error) {
+			var tempNightwatchConfigPath = nightwatchConfigPath.replace('json', 'tmp.json')
```
Comment (samgiles) [inline lib/tasks/test.js]: There is a JSHint error here: missing semi colon.

**Comment by samgiles** [comment]
Nice, :+1: merge when ready. :)
